celery/backends/redis.py

```diff
@@ -449,27 +452,47 @@ def on_chord_part_return(self, request, state, result,
             callback = maybe_signature(request.chord, app=app)
             total = callback['chord_size'] + totaldiff
             if readycount == total:
-                decode, unpack = self.decode, self._unpack_chord_result
-                with client.pipeline() as pipe:
-                    if self._chord_zset:
-                        pipeline = pipe.zrange(jkey, 0, -1)
-                    else:
-                        pipeline = pipe.lrange(jkey, 0, total)
-                    resl, = pipeline.execute()
-                try:
-                    callback.delay([unpack(tup, decode) for tup in resl])
+                hr = GroupResult.restore(gid)
```

Comment:
full form of hr variable would be more readable

Reply:
This is now `header_result`